<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started with 'SpotSweeper' • SpotSweeper</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with 'SpotSweeper'">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SpotSweeper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/getting_started.html">Getting started with 'SpotSweeper'</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/MicTott/SpotSweeper/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Getting started with 'SpotSweeper'</h1>
                        <h4 data-toc-skip class="author">Michael
Totty</h4>
            <address class="author_afil">
      Johns Hopkins Bloomberg School of Public Health, Baltimore, MD,
USA<br><h4 data-toc-skip class="author">Boyi
Guo</h4>
            <address class="author_afil">
      Johns Hopkins Bloomberg School of Public Health, Baltimore, MD,
USA<br><h4 data-toc-skip class="author">Stephanie
Hicks</h4>
            <address class="author_afil">
      Johns Hopkins Bloomberg School of Public Health, Baltimore, MD,
USA<br><h4 data-toc-skip class="date">2024-08-17</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/MicTott/SpotSweeper/blob/devel/vignettes/getting_started.Rmd" class="external-link"><code>vignettes/getting_started.Rmd</code></a></small>
      <div class="d-none name"><code>getting_started.Rmd</code></div>
    </address>
</address>
</address>
</div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>SpotSweeper</code> is an R package for spatial transcriptomics
data quality control (QC). It provides functions for detecting and
visualizing spot-level local outliers and artifacts using
spatially-aware methods. The package is designed to work with <a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a>
objects, and is compatible with data from 10X Genomics Visium and other
spatial transcriptomics platforms.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Currently, the only way to install <code>SpotSweeper</code> is by
downloading the development version which can be installed from <a href="https://github.com/MicTott/SpotSweeper" class="external-link">GitHub</a> using the
following:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"MicTott/SpotSweeper"</span><span class="op">)</span></span></code></pre></div>
<p>Once accepted in <a href="http://bioconductor.org/" class="external-link">Bioconductor</a>,
<code>SpotSweeper</code> will be installable using:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"SpotSweeper"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spot-level-local-outlier-detection">Spot-level local outlier detection<a class="anchor" aria-label="anchor" href="#spot-level-local-outlier-detection"></a>
</h2>
<div class="section level3">
<h3 id="loading-example-data">Loading example data<a class="anchor" aria-label="anchor" href="#loading-example-data"></a>
</h3>
<p>Here we’ll walk you through the standard workflow for using
‘SpotSweeper’ to detect and visualize local outliers in spatial
transcriptomics data. We’ll use the <code>Visium_humanDLPFC</code>
dataset from the <code>STexampleData</code> package, which is a
<code>SpatialExperiment</code> object.</p>
<p>Because local outliers will be saved in the <code>colData</code> of
the <code>SpatialExperiment</code> object, we’ll first view the
<code>colData</code> and drop out-of-tissue spots before calculating
quality control (QC) metrics and running <code>SpotSweeper</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MicTott/SpotSweeper" class="external-link">SpotSweeper</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load  Maynard et al DLPFC daatset</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">STexampleData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/STexampleData/man/STexampleData.html" class="external-link">Visium_humanDLPFC</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?STexampleData and browseVignettes('STexampleData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show column data before SpotSweeper</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "barcode_id"   "sample_id"    "in_tissue"    "array_row"    "array_col"   </span></span>
<span><span class="co">## [6] "ground_truth" "reference"    "cell_count"</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># drop out-of-tissue spots</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>, <span class="va">spe</span><span class="op">$</span><span class="va">in_tissue</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="calculating-qc-metrics-using-scuttle">Calculating QC metrics using <code>scuttle</code><a class="anchor" aria-label="anchor" href="#calculating-qc-metrics-using-scuttle"></a>
</h3>
<p>We’ll use the <code>scuttle</code> package to calculate QC metrics.
To do this, we’ll need to first change the <code>rownames</code> from
gene id to gene names. We’ll then get the mitochondrial transcripts and
calculate QC metrics for each spot using
<code><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html" class="external-link">scuttle::addPerCellQCMetrics</a></code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># change from gene id to gene names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span></span>
<span></span>
<span><span class="co"># identifying the mitochondrial transcripts</span></span>
<span><span class="va">is.mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"^MT-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># calculating QC metrics for each spot using scuttle</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">scuttle</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html" class="external-link">addPerCellQCMetrics</a></span><span class="op">(</span><span class="va">spe</span>, subsets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Mito <span class="op">=</span> <span class="va">is.mito</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "barcode_id"            "sample_id"             "in_tissue"            </span></span>
<span><span class="co">##  [4] "array_row"             "array_col"             "ground_truth"         </span></span>
<span><span class="co">##  [7] "reference"             "cell_count"            "sum"                  </span></span>
<span><span class="co">## [10] "detected"              "subsets_Mito_sum"      "subsets_Mito_detected"</span></span>
<span><span class="co">## [13] "subsets_Mito_percent"  "total"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="identifying-local-outliers-using-spotsweeper">Identifying local outliers using <code>SpotSweeper</code><a class="anchor" aria-label="anchor" href="#identifying-local-outliers-using-spotsweeper"></a>
</h3>
<p>We can now use <code>SpotSweeper</code> to identify local outliers in
the spatial transcriptomics data. We’ll use the
<code>localOutliers</code> function to detect local outliers based on
the unique detected genes, total library size, and percent of the total
reads that are mitochondrial. These methods assume a normal
distribution, so we’ll use the log-transformed sum of the counts and the
log-transformed number of detected genes. For mitochondrial percent,
we’ll use the raw mitochondrial percentage.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># library size</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/localOutliers.html">localOutliers</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"sum"</span>,</span>
<span>    direction <span class="op">=</span> <span class="st">"lower"</span>,</span>
<span>    log <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># unique genes</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/localOutliers.html">localOutliers</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"detected"</span>,</span>
<span>    direction <span class="op">=</span> <span class="st">"lower"</span>,</span>
<span>    log <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># mitochondrial percent</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/localOutliers.html">localOutliers</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"subsets_Mito_percent"</span>,</span>
<span>    direction <span class="op">=</span> <span class="st">"higher"</span>,</span>
<span>    log <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The <code>localOutlier</code> function automatically outputs the
results to the <code>colData</code> with the naming convention
<code>X_outliers</code>, where <code>X</code> is the name of the input
<code>colData</code>. We can then combine all outliers into a single
column called <code>local_outliers</code> in the <code>colData</code> of
the <code>SpatialExperiment</code> object.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># combine all outliers into "local_outliers" column</span></span>
<span><span class="va">spe</span><span class="op">$</span><span class="va">local_outliers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/logical.html" class="external-link">as.logical</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">sum_outliers</span><span class="op">)</span> <span class="op">|</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/logical.html" class="external-link">as.logical</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">detected_outliers</span><span class="op">)</span> <span class="op">|</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/logical.html" class="external-link">as.logical</a></span><span class="op">(</span><span class="va">spe</span><span class="op">$</span><span class="va">subsets_Mito_percent_outliers</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualizing-local-outliers">Visualizing local outliers<a class="anchor" aria-label="anchor" href="#visualizing-local-outliers"></a>
</h3>
<p>We can visualize the local outliers using the <code>plotQC</code>
function. This function creates a scatter plot of the specified metric
and highlights the local outliers in red using the <code>escheR</code>
package. Here, we’ll visualize local outliers of library size, unique
genes, mitochondrial percent, and finally, all local outliers. We’ll
then arrange these plots in a grid using
<code>ggpubr::arrange</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boyiguo1/escheR" class="external-link">escheR</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: ggplot2</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># library size</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"sum_log"</span>,</span>
<span>    outliers <span class="op">=</span> <span class="st">"sum_outliers"</span>, point_size <span class="op">=</span> <span class="fl">1.1</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Library Size"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># unique genes</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"detected_log"</span>,</span>
<span>    outliers <span class="op">=</span> <span class="st">"detected_outliers"</span>, point_size <span class="op">=</span> <span class="fl">1.1</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Unique Genes"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># mitochondrial percent</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"subsets_Mito_percent"</span>,</span>
<span>    outliers <span class="op">=</span> <span class="st">"subsets_Mito_percent_outliers"</span>, point_size <span class="op">=</span> <span class="fl">1.1</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Mitochondrial Percent"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># all local outliers</span></span>
<span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"sum_log"</span>,</span>
<span>    outliers <span class="op">=</span> <span class="st">"local_outliers"</span>, point_size <span class="op">=</span> <span class="fl">1.1</span>, stroke <span class="op">=</span> <span class="fl">0.75</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"All Local Outliers"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot</span></span>
<span><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span></span>
<span>    plotlist <span class="op">=</span> <span class="va">plot_list</span>,</span>
<span>    ncol <span class="op">=</span> <span class="fl">2</span>, nrow <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    common.legend <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="getting_started_files/figure-html/local_outlier_plot-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="removing-technical-artifacts-using-spotsweeper">Removing technical artifacts using <code>SpotSweeper</code><a class="anchor" aria-label="anchor" href="#removing-technical-artifacts-using-spotsweeper"></a>
</h2>
<div class="section level3">
<h3 id="loading-example-data-1">Loading example data<a class="anchor" aria-label="anchor" href="#loading-example-data-1"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load in DLPFC sample with hangnail artifact</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">DLPFC_artifact</span><span class="op">)</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">DLPFC_artifact</span></span>
<span></span>
<span><span class="co"># inspect colData before artifact detection</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "sample_id"          "in_tissue"          "array_row"         </span></span>
<span><span class="co">##  [4] "array_col"          "key"                "sum_umi"           </span></span>
<span><span class="co">##  [7] "sum_gene"           "expr_chrM"          "expr_chrM_ratio"   </span></span>
<span><span class="co">## [10] "ManualAnnotation"   "subject"            "region"            </span></span>
<span><span class="co">## [13] "sex"                "age"                "diagnosis"         </span></span>
<span><span class="co">## [16] "sample_id_complete" "count"              "sizeFactor"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="visualizing-technical-artifacts">Visualizing technical artifacts<a class="anchor" aria-label="anchor" href="#visualizing-technical-artifacts"></a>
</h3>
<p>Technical artifacts can commonly be visualized by standard QC
metrics, including library size, unique genes, or mitochondrial
percentage. We can first visualize the technical artifacts using the
<code>plotQC</code> function. This function plots the Visium spots with
the specified QC metric.We’ll then again arrange these plots using
<code>ggpubr::arrange</code>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># library size</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"sum_umi"</span>,</span>
<span>    outliers <span class="op">=</span> <span class="cn">NULL</span>, point_size <span class="op">=</span> <span class="fl">1.1</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Library Size"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># unique genes</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"sum_gene"</span>,</span>
<span>    outliers <span class="op">=</span> <span class="cn">NULL</span>, point_size <span class="op">=</span> <span class="fl">1.1</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Unique Genes"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># mitochondrial percent</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"expr_chrM_ratio"</span>,</span>
<span>    outliers <span class="op">=</span> <span class="cn">NULL</span>, point_size <span class="op">=</span> <span class="fl">1.1</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Mitochondrial Percent"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot</span></span>
<span><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span></span>
<span>    plotlist <span class="op">=</span> <span class="va">plot_list</span>,</span>
<span>    ncol <span class="op">=</span> <span class="fl">3</span>, nrow <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    common.legend <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="getting_started_files/figure-html/artifact_QC_plots-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="identifying-artifacts-using-spotsweeper">Identifying artifacts using <code>SpotSweeper</code><a class="anchor" aria-label="anchor" href="#identifying-artifacts-using-spotsweeper"></a>
</h3>
<p>We can then use the <code>findArtifacts</code> function to identify
artifacts in the spatial transcriptomics (data. This function identifies
technical artifacts based on the first principle component of the local
variance of the specified QC metric (<code>mito_percent</code>) at
numerous neighorhood sizes (<code>n_rings=5</code>). Currently,
<code>kmeans</code> clustering is used to cluster the technical artifact
vs high-quality Visium spots. Similar to <code>localOutliers</code>, the
<code>findArtifacts</code> function then outputs the results to the
<code>colData</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># find artifacts using SpotSweeper</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findArtifacts.html">findArtifacts</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    mito_percent <span class="op">=</span> <span class="st">"expr_chrM_ratio"</span>,</span>
<span>    mito_sum <span class="op">=</span> <span class="st">"expr_chrM"</span>,</span>
<span>    n_rings <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    name <span class="op">=</span> <span class="st">"artifact"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># check that "artifact" is now in colData</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "sample_id"           "in_tissue"           "array_row"          </span></span>
<span><span class="co">##  [4] "array_col"           "key"                 "sum_umi"            </span></span>
<span><span class="co">##  [7] "sum_gene"            "expr_chrM"           "expr_chrM_ratio"    </span></span>
<span><span class="co">## [10] "ManualAnnotation"    "subject"             "region"             </span></span>
<span><span class="co">## [13] "sex"                 "age"                 "diagnosis"          </span></span>
<span><span class="co">## [16] "sample_id_complete"  "count"               "sizeFactor"         </span></span>
<span><span class="co">## [19] "expr_chrM_ratio_log" "coords"              "k6"                 </span></span>
<span><span class="co">## [22] "k18"                 "k36"                 "k60"                </span></span>
<span><span class="co">## [25] "k90"                 "artifact"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="visualizing-artifacts">Visualizing artifacts<a class="anchor" aria-label="anchor" href="#visualizing-artifacts"></a>
</h3>
<p>We can visualize the artifacts using the <code>escheR</code> package.
Here, we’ll visualize the artifacts using the <code>plotQC</code>
function and arrange these plots using <code>ggpubr::arrange</code>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotQC.html">plotQC</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"expr_chrM_ratio"</span>,</span>
<span>    outliers <span class="op">=</span> <span class="st">"artifact"</span>, point_size <span class="op">=</span> <span class="fl">1.1</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Hangnail artifact"</span><span class="op">)</span></span></code></pre></div>
<p><img src="getting_started_files/figure-html/artifact_visualization-1.png" width="700">
# Session information</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ggpubr_0.6.0                escheR_1.4.0               </span></span>
<span><span class="co">##  [3] ggplot2_3.5.1               STexampleData_1.12.3       </span></span>
<span><span class="co">##  [5] SpatialExperiment_1.14.0    SingleCellExperiment_1.26.0</span></span>
<span><span class="co">##  [7] SummarizedExperiment_1.34.0 Biobase_2.64.0             </span></span>
<span><span class="co">##  [9] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">## [11] IRanges_2.38.1              S4Vectors_0.42.1           </span></span>
<span><span class="co">## [13] MatrixGenerics_1.16.0       matrixStats_1.3.0          </span></span>
<span><span class="co">## [15] ExperimentHub_2.12.0        AnnotationHub_3.12.0       </span></span>
<span><span class="co">## [17] BiocFileCache_2.12.0        dbplyr_2.5.0               </span></span>
<span><span class="co">## [19] BiocGenerics_0.50.0         SpotSweeper_1.1.0          </span></span>
<span><span class="co">## [21] BiocStyle_2.32.1           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] DBI_1.2.3                 rlang_1.1.4              </span></span>
<span><span class="co">##  [3] magrittr_2.0.3            spatialEco_2.0-2         </span></span>
<span><span class="co">##  [5] compiler_4.4.1            RSQLite_2.3.7            </span></span>
<span><span class="co">##  [7] DelayedMatrixStats_1.26.0 png_0.1-8                </span></span>
<span><span class="co">##  [9] systemfonts_1.1.0         vctrs_0.6.5              </span></span>
<span><span class="co">## [11] pkgconfig_2.0.3           crayon_1.5.3             </span></span>
<span><span class="co">## [13] fastmap_1.2.0             backports_1.5.0          </span></span>
<span><span class="co">## [15] magick_2.8.4              XVector_0.44.0           </span></span>
<span><span class="co">## [17] labeling_0.4.3            scuttle_1.14.0           </span></span>
<span><span class="co">## [19] utf8_1.2.4                rmarkdown_2.27           </span></span>
<span><span class="co">## [21] UCSC.utils_1.0.0          ragg_1.3.2               </span></span>
<span><span class="co">## [23] purrr_1.0.2               bit_4.0.5                </span></span>
<span><span class="co">## [25] xfun_0.46                 beachmat_2.20.0          </span></span>
<span><span class="co">## [27] zlibbioc_1.50.0           cachem_1.1.0             </span></span>
<span><span class="co">## [29] jsonlite_1.8.8            blob_1.2.4               </span></span>
<span><span class="co">## [31] highr_0.11                DelayedArray_0.30.1      </span></span>
<span><span class="co">## [33] BiocParallel_1.38.0       terra_1.7-78             </span></span>
<span><span class="co">## [35] broom_1.0.6               parallel_4.4.1           </span></span>
<span><span class="co">## [37] R6_2.5.1                  bslib_0.8.0              </span></span>
<span><span class="co">## [39] car_3.1-2                 jquerylib_0.1.4          </span></span>
<span><span class="co">## [41] Rcpp_1.0.13               bookdown_0.40            </span></span>
<span><span class="co">## [43] knitr_1.48                Matrix_1.7-0             </span></span>
<span><span class="co">## [45] tidyselect_1.2.1          abind_1.4-5              </span></span>
<span><span class="co">## [47] yaml_2.3.10               codetools_0.2-20         </span></span>
<span><span class="co">## [49] curl_5.2.1                lattice_0.22-6           </span></span>
<span><span class="co">## [51] tibble_3.2.1              withr_3.0.1              </span></span>
<span><span class="co">## [53] KEGGREST_1.44.1           evaluate_0.24.0          </span></span>
<span><span class="co">## [55] desc_1.4.3                Biostrings_2.72.1        </span></span>
<span><span class="co">## [57] pillar_1.9.0              BiocManager_1.30.23      </span></span>
<span><span class="co">## [59] filelock_1.0.3            carData_3.0-5            </span></span>
<span><span class="co">## [61] generics_0.1.3            BiocVersion_3.19.1       </span></span>
<span><span class="co">## [63] sparseMatrixStats_1.16.0  munsell_0.5.1            </span></span>
<span><span class="co">## [65] scales_1.3.0              glue_1.7.0               </span></span>
<span><span class="co">## [67] tools_4.4.1               BiocNeighbors_1.22.0     </span></span>
<span><span class="co">## [69] ggsignif_0.6.4            fs_1.6.4                 </span></span>
<span><span class="co">## [71] cowplot_1.1.3             grid_4.4.1               </span></span>
<span><span class="co">## [73] tidyr_1.3.1               AnnotationDbi_1.66.0     </span></span>
<span><span class="co">## [75] colorspace_2.1-1          GenomeInfoDbData_1.2.12  </span></span>
<span><span class="co">## [77] cli_3.6.3                 rappdirs_0.3.3           </span></span>
<span><span class="co">## [79] textshaping_0.4.0         fansi_1.0.6              </span></span>
<span><span class="co">## [81] viridisLite_0.4.2         S4Arrays_1.4.1           </span></span>
<span><span class="co">## [83] dplyr_1.1.4               gtable_0.3.5             </span></span>
<span><span class="co">## [85] rstatix_0.7.2             sass_0.4.9               </span></span>
<span><span class="co">## [87] digest_0.6.36             SparseArray_1.4.8        </span></span>
<span><span class="co">## [89] farver_2.1.2              rjson_0.2.21             </span></span>
<span><span class="co">## [91] memoise_2.0.1             htmltools_0.5.8.1        </span></span>
<span><span class="co">## [93] pkgdown_2.1.0             lifecycle_1.0.4          </span></span>
<span><span class="co">## [95] httr_1.4.7                mime_0.12                </span></span>
<span><span class="co">## [97] bit64_4.0.5               MASS_7.3-60.2</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Michael Totty, Boyi Guo.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
